pkg/controller/networkpolicy/mutate.go

```diff
@@ -0,0 +1,200 @@
+				klog.Errorf("Error de-serializing old Antrea ClusterNetworkPolicy")
+				return GetAdmissionResponseForErr(err)
+			}
+		}
+		msg, allowed, patch = m.mutateAntreaPolicyRuleName(op, curACNP.Spec.Ingress, curACNP.Spec.Egress)
+	case "NetworkPolicy":
+		klog.V(2).Info("Mutating Antrea NetworkPolicy CRD")
+		var curANP, oldANP secv1alpha1.NetworkPolicy
+		if curRaw != nil {
+			if err := json.Unmarshal(curRaw, &curANP); err != nil {
+				klog.Errorf("Error de-serializing current Antrea NetworkPolicy")
+				return GetAdmissionResponseForErr(err)
+			}
+		}
+		if oldRaw != nil {
+			if err := json.Unmarshal(oldRaw, &oldANP); err != nil {
+				klog.Errorf("Error de-serializing old Antrea NetworkPolicy")
+				return GetAdmissionResponseForErr(err)
+			}
+		}
+		msg, allowed, patch = m.mutateAntreaPolicyRuleName(op, curANP.Spec.Ingress, curANP.Spec.Egress)
+	}
+
+	if msg != "" {
+		result = &metav1.Status{
+			Message: msg,
+		}
+	}
+	return &admv1.AdmissionResponse{
+		Allowed:   allowed,
+		Result:    result,
+		PatchType: &patchType,
+		Patch:     patch,
+	}
+}
+
+// mutateAntreaPolicyRuleName mutates names of rules of an Antrea NetworkPolicy CRD.
+// If users didn't specify the name of an ingress or egress rule,
+// mutateAntreaPolicyRuleName will auto-generate a name for this rule.
+func (m *NetworkPolicyMutator) mutateAntreaPolicyRuleName(op admv1.Operation, ingress, egress []secv1alpha1.Rule) (string, bool, []byte) {
+	allowed := true
+	reason := ""
+	var patch []byte
+	switch op {
+	case admv1.Create, admv1.Update:
+		ruleNames := sets.NewString()
+		for _, rule := range append(ingress, egress...) {
+			ruleNames.Insert(rule.Name)
+		}
+
+		ingressRulePaths, ingressRuleNames := generateRuleNames(ruleNames, "ingress", ingress)
+		egressRulePaths, egressRuleNames := generateRuleNames(ruleNames, "egress", egress)
+
+		genPatch, err := createReplacePatch(append(ingressRulePaths, egressRulePaths...), append(ingressRuleNames, egressRuleNames...))
+		if err != nil {
+			allowed = false
+			reason = "unable to generate mutating patch"
+			break
+		}
+		patch = genPatch
+
+	case admv1.Delete:
+		// Delete of Antrea Policies have no mutation
+		allowed = true
+	}
+	return reason, allowed, patch
+}
+
+// generateRuleNames generates unique rule names and returns a list of json paths and the corresponding list of generated names
+func generateRuleNames(ruleNames sets.String, prefix string, rules []secv1alpha1.Rule) ([]string, []string) {
+	var paths []string
+	var values []string
+	for idx, rule := range rules {
+		if rule.Name == "" {
+			genName := fmt.Sprintf("%s-%s-%s", prefix, strings.ToLower(string(*rule.Action)), hashRule(rule))
+			// If an existing name is generated, it means there are duplicate rules or
+			// one of rule-names contains hash value of other rules, which is illegal.
+			// MutatingWebhook will skip auto-gen of this rule as a signal to pass illegal input info to ValidatingWebhook
```

Comment:
"MutatingWebhook will skip auto-gen of this rule name as a signal to ValidatingWebhook that an error needs to be reported"

Reply:
Thx!